File: backend/app/schemas/position.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class PositionBase(BaseModel):
    """Базовая схема должности"""
    name: str
    code: str
    division_id: Optional[int] = None
    section_id: Optional[int] = None
    attribute: Optional[str] = None
    description: Optional[str] = None
    is_active: bool = True
    
class PositionCreate(PositionBase):
    """Схема для создания должности"""
    function_ids: Optional[List[int]] = []
    
class PositionUpdate(BaseModel):
    """Схема для обновления должности"""
    name: Optional[str] = None
    code: Optional[str] = None
    division_id: Optional[int] = None
    section_id: Optional[int] = None
    attribute: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    function_ids: Optional[List[int]] = None
    
class PositionInDBBase(PositionBase):
    """Базовая схема для должности в БД"""
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        
class Position(PositionInDBBase):
    """Схема для возврата должности через API"""
    function_ids: List[int] = [] 

File: frontend/src/types/position.ts
import { Section } from './section';
import { Division } from './division';

export interface Position {
  id: number;
  name: string;
  code: string;
  division_id: number | null;
  section_id: number | null;
  attribute: string | null;
  description: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  section?: Section; // Опциональный объект отдела
  division?: Division; // Опциональный объект подразделения
  function_ids?: number[]; // Массив ID связанных функций
}

export interface PositionCreate {
  name: string;
  code: string;
  section_id?: number;
  division_id?: number | null;
  attribute?: string | null;
  description?: string | null;
  is_active?: boolean;
  function_ids?: number[]; // Массив ID связанных функций
}

export type PositionUpdate = Partial<PositionCreate>; 